internal/attestation/crafter/crafter.go

```diff
@@ -372,49 +372,43 @@ func persistCraftingState(craftState *api.CraftingState, stateFilePath string) e
 
 // ResolveEnvVars will iterate on the env vars in the allow list and resolve them from the system context
 // strict indicates if it should fail if any env variable can not be found
-func (c *Crafter) ResolveEnvVars(strict bool) error {
+func (c *Crafter) ResolveEnvVars() error {
 	if err := c.requireStateLoaded(); err != nil {
 		return err
 	}
 
-	// Runner specific env variables
+	// Runner specific environment variables
+
 	c.logger.Debug().Str("runnerType", c.Runner.String()).Msg("loading runner specific env variables")
-	var outputEnvVars = make(map[string]string)
 	if !c.Runner.CheckEnv() {
 		errorStr := fmt.Sprintf("couldn't detect the environment %q. Is the crafting process happening in the target env?", c.Runner.String())
-		if strict {
-			return fmt.Errorf("%s - %w", errorStr, ErrRunnerContextNotFound)
-		}
-		c.logger.Warn().Msg(errorStr)
-	} else {
-		c.logger.Debug().Str("runnerType", c.Runner.String()).Strs("variables", c.Runner.ListEnvVars()).Msg("list of env variables to automatically extract")
-		outputEnvVars = c.Runner.ResolveEnvVars()
-		if notFound := notResolvedVars(outputEnvVars, c.Runner.ListEnvVars()); len(notFound) > 0 {
-			if strict {
-				return fmt.Errorf("required env variables not present %q", notFound)
-			}
-			c.logger.Warn().Strs("key", notFound).Msg("required env variables not present")
-		}
+		return fmt.Errorf("%s - %w", errorStr, ErrRunnerContextNotFound)
 	}
 
-	// User-defined env vars
-	varsAllowList := c.CraftingState.InputSchema.EnvAllowList
-	if len(varsAllowList) > 0 {
-		c.logger.Debug().Strs("allowList", varsAllowList).Msg("loading env variables")
-		for _, want := range varsAllowList {
-			val := os.Getenv(want)
-			if val == "" {
-				continue
-			}
+	// Workflow run environment variables
 
-			outputEnvVars[want] = val
-		}
+	varNames := make([]string, len(c.Runner.ListEnvVars()))
+	for index, envVarDef := range c.Runner.ListEnvVars() {
```

Comment:
Absolutely.

If you append, you are reallocating memory dynamically every time the capacity of the slice goes to zero (which happens frequently at the beginning and less frequently as you go).

If you do it the way I did, you allocate memory only once. This is convenient in this case, performance-wise, because we know exactly the amount of cells we need.

Depending on the environment, this could also generate a performance linter. I don't remember if I saw this here or somewhere else, but I think it happened here as well.

Reply:
To preallocate, can't you define a slice with capacity 3 and initial len 0?

```
varNames := make([]string, 0, 3)
```

https://go.dev/play/p/KSFZkLVYmZP

Reply:
Yes, that would work as well. Both methods achieve the same goal, I think